src/model/destinations-model.js

```diff
@@ -20,4 +25,14 @@ export default class DestinationsModel {
     return this.#destinations.map((item) => item.name);
   }
 
+  async init() {
+    try {
+      const destinations = await this.#destinationsApiService.destinations;
+      this.#destinations = destinations;
+    } catch(err) {
+      this.#destinations = [];
+    }
+    this._notify(UpdateType.INIT);
+  }
```

Comment:
перенеси нотифаи лучше в finally блок, что бы обновление тригерилось только когда данные придут 
```suggestion
    try {
      const destinations = await this.#destinationsApiService.destinations;
      this.#destinations = destinations;
    } catch(err) {
      this.#destinations = [];
    } finally {
       this._notify(UpdateType.INIT);
    }
  }
```

Reply:
ну и сейчас ошибки вылетают, они как раз и связаны с тем что у тебя вызывается обновление из одной модели когда дынные в другую ещё не пришли, так что везде где вылетает, добавь проверки, на то есть ли данные, что бы код выполнялся без ошибок, а как данные придут там автоматически всё перерисуется